subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -158,6 +158,11 @@ public String toString() {
         return versionName;
     }
 
+    // We have to keep this for a while: https://github.com/gradle/gradle/issues/4856
```

Comment:
Sorry for commenting on a PR this old, but I just noted this comment in the Gradle sources.
The usage of this method has been removed almost 3 years, so I guess that by now you could get rid of it: https://github.com/JetBrains/intellij-community/commit/014afeb2d8587b0dfd7d4c82d56891ea708bbdb0

Just a small hint, nothing more 😉

Reply:
Thanks! After six years we'll remove this in https://github.com/gradle/gradle/pull/28825. I guess that qualifies as "for a while". :)